shopgate/pwa
# Implemented handling of "registrationMode" shop setting

# Description
This ticket is about to implement the "registrationMode" setting into the PWA. The setting has two possible values right now: "simple" and "extended".

At mode "extended" the registration form look and behave like usual. When mode "simple" is active, the registration form only contains first name, last name, credentials and the customer attributes if configured. When this mode is active, only basic customers are created without addresses. Addresses are supposed to be created within the checkout process.

This ticket also updates the form styling within theme-ios11. From now on forms are presented in Material Design styling line used for the theme-gmd theme.

## Type of change


## How to test it
The current registration mode is determined via the getRegistrationMode selector within @shopgate/engage/core/selectors/shopSettings . To test the different behaviours you either need to change the "registrationMode" shop setting for you test shop, or mock the return value of the selector.

When mode "extended" is active, the registration forms should look and behave as before. You need to test the registration via the regular "Register" link, and at transition between cart and checkout with an "direct ship" item in the cart, since direct ship carts enable a toggle to show a special shipping address form.

When mode "simple" is active, the billing form should not be visible anymore, but name and last name are shown within the "Create Account" form. Additionally the shipping address toggle should not come up. Registration should work in any case, also server side form validation needs still to work as expected.

For now the feature ist not active for "Guest Registration" since processes differ from the regular registration and need full  addresses.
